bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -1860,7 +1863,11 @@ bool BCStateTran::onMessage(const CheckpointSummaryMsg *m, uint32_t msgLen, uint
       totalBlocksLeftToCollectInCycle_ = maxBlockIdToCollectInCycle_ - minBlockIdToCollectInCycle_ + 1;
     } else {
       // fetch reserved pages (vblock)
-      ConcordAssertEQ(newCheckpoint.maxBlockId, lastReachableBlockNum);
+      if (newCheckpoint.maxBlockId < lastReachableBlockNum) {
+        // In some rare cases replica might need to collect reserved pages which it already has (this is done that way
```

Comment:
can we explain what those rare cases are?

Reply:
I wanted to, but thought it would be too complicated and too much. 
I'll have to explain in few words (shortly). Explaining the full scenario will require a confluence page and I don't think it makes sense to do so.